### PR TITLE
Provisioner updates

### PR DESCRIPTION
**Pull request by jhford**
I got so sick of remembering how the old boilerplate and config stuff worked so I decided to bring the provisioner up to date.  This is the first step in changing which azure account we use.

### CODE REVIEW

**Comment by jonasfj** [comment]
YAY!!!!

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    },
+  },
+
+  validator: {
+    requires: ['cfg'],
+    setup: async ({cfg}) => {
+      return await base.validator({
+        folder: path.join(__dirname, '..', 'schemas'),
+        constants: require('../schemas/constants'),
+        publish: cfg.app.publishMetaData === 'true',
```
Comment (jonasfj) [inline src/main.js]: newest validator guess this using `NODE_ENV == 'production'`... I think...

@imbstack, can you help review this main.js, specifically, you are 100% sure about what is best practice and not...

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+  provisioner: {
+    requires: ['cfg', 'awsManager', 'WorkerType', 'WorkerState', 'Secret', 'ec2', 'influx'],
+    setup: async ({cfg, awsManager, WorkerType, WorkerState, Secret, ec2, influx}) => {
+      let queue = new taskcluster.Queue({credentials: cfg.taskcluster.credentials});
+
+      let provisioner = new provision.Provisioner({
+        WorkerType: WorkerType,
+        Secret: Secret,
+        WorkerState: WorkerState,
+        queue: queue,
```
Comment (jonasfj) [inline src/main.js]: It's probably easier to write and read... if you just do: {WorkerType, Secret, WorkerState, queue, provisionerId: cfg.app.id, ....

but also a nit...

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+        awsManager: awsManager,
+        provisionIterationInterval: cfg.app.iterationInterval,
+        dmsApiKey: cfg.deadmanssnitch.api.key,
+        iterationSnitch: cfg.deadmanssnitch.iterationSnitch,
+      });
+
+      try {
+        provisioner.run();
+      } catch (err) {
+        debug('[alert-operator] Error: ' + err.stack || err);
```
Comment (jonasfj) [inline src/main.js]: monitor.reportError(err) will make your life so much better :)

Comment (jhford) [inline src/main.js]: yup!  need to figure out the stats stuff as it relates to the biaser before I rip out lib-stats and put in lib-monitor.

File: config.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
+defaults:
+  app:
+    publishMetaData: false
```
Comment (jonasfj) [inline config.yml]: `!env:bool PUBLISH_METADATA` is what I would use...

Maybe compare to: https://github.com/taskcluster/taskcluster-queue/blob/master/config.yml

Comment (jonasfj) [inline config.yml]: it defaults to false...

File: config.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    publicUrl: https://localhost:60407
+    port: !env:number PORT
+    env: test
+    forceSSL: false
+    trustProxy: false
+  deadmanssnitch:
+    api:
+      key: 'nokey'
+    iterationSnitch: 'http://httpbin.org/status/200'
+
```
Comment (jonasfj) [inline config.yml]: In `test` please set the `azure.account = "jungle"` this is the garbage account...
You test environment shouldn't have scopes to talk to the production account...

File: config.yml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    iterationInterval: 10000
+    maxInstanceLife: '- 96 hours'
+    id: aws-provisioner-dev
+    workerTypeTableName: DevWorkerTypes
+    secretTableName: DevWorkerSecrets
+    workerStateTableName: DevWorkerState
+    awsInstancePubkey: !env AWS_INSTANCE_PUBKEY
+    awsKeyPrefix: "aws-provisioner-dev-managed:"
+    statsComponent: aws-provisioner-dev
+    allowedRegions: "us-west-2,us-east-1,us-west-1"
```
Comment (jonasfj) [inline config.yml]: `!env:list ALLOWED_REGIONS`

Then configure with ALLOWED_REGIONS="us-west-2 us-east-1 us-west-1"
If you prefer to hard code it,... then just do that in the `production` section...
`allowedRegions: ['us-west-2', 'us-east-1', 'us-west-1']` or even use YAML-style lists :)

Comment (jonasfj) [inline config.yml]: This way you don't have to do any string splitting in-tree...

File: user-config-example.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+defaults:
+  aws:
+    accessKeyId:        '...'
+    secretAccessKey:    '...'
+    region:             us-west-2
+  azure:
+    accountName:        jungle
```
Comment (jonasfj) [inline user-config-example.yml]: YAY, remove the `accountKey` entry... no need... you can easily get jungle access using tc credentials..

**Comment by jonasfj** [comment]
# `r+`

I think my comments are minor... from skimming it generally seems okay... Definitely better than what we had before...

File: src/main.js (hunk 2 above)
Comment (jhford) [inline src/main.js]: i guess I'm weird, because I really find that syntax way less clear.

Comment (jonasfj) [inline src/main.js]: okay, that's fine... I can totally live with this...

Personally, I find this without alignment, hard to read... on the plus side I do see that it forces me to read it more closely :)

Anyways, it's not remotely important... I just wanted to highlight that you had options...

**Pull request merged**